flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/LeastUtilizationSlotMatchingStrategy.java

```diff
@@ -79,4 +94,19 @@ private static double calculateUtilization(
 
         return (double) (numberRegisteredSlots - numberFreeSlots) / numberRegisteredSlots;
     }
+
+    private static double calculateUtilization(
+            TaskManagerResourceInfoSnapshot instanceResourceInfoSnapshot) {
+        if (instanceResourceInfoSnapshot.getTotalProfile() == ResourceProfile.UNKNOWN
+                || instanceResourceInfoSnapshot.getAvailableProfile() == ResourceProfile.UNKNOWN) {
+            return instanceResourceInfoSnapshot.getAllocatedSlotNumber();
+        } else {
+            return instanceResourceInfoSnapshot
+                    .getTotalProfile()
+                    .getCpuCores()
```

Comment:
It seems that we only consider Cpu utilization for fine-grained resource. Do we need to consider other dimensions, such as memory?
If the first version only considers cpu resource for the time being, we should at least add a comment to explain why we choose this strategy now.